source/priority-queue.ts

```diff
@@ -34,6 +34,10 @@ export default class PriorityQueue implements Queue<PriorityQueueOptions> {
 		return item && item.run;
 	}
 
+	sizeBy(options: Partial<PriorityQueueOptions>): number {
+		return this._queue.filter(element => element.priority === options.priority).length;
+	}
```

Comment:
Maybe the queue itself would only expose a `filter` method? That way it would be simpler to implement custom queues and `sizeBy()` logic is contained in `pQueue` itself and not the underlying queue.